scipy/stats/stats.py

```diff
@@ -2306,6 +2311,186 @@ def zmap(scores, compare, axis=0, ddof=0):
         return (scores - mns) / sstd
 
 
+def _iqr_scale_to_num(scale):
+    """
+    A helper function to convert the 'scale' argument of `iqr` to a number.
+
+    scale: str
+        If scale is one of ['raw', 'normal'], this function will map the string
+        to a value.
+
+    The concept for this function is borrowed heavily from the `numpy.histogram`
+    helper function `_hist_optim_numbins_estimator`.
+    """
+    assert isinstance(scale, string_types)
+    # private function should not be called otherwise
+
+    # See https://en.wikipedia.org/wiki/Robust_measures_of_scale
+    scale_conversions = {'raw': 1.0,
+                         'normal': special.erfinv(0.5) * 2.0 * math.sqrt(2.0)}
+    try:
+        return scale_conversions[scale.lower()]
+    except KeyError:
+        raise ValueError("{0} not a valid scale for `iqr`".format(scale))
+
+
+def iqr(x, axis=None, range=(25, 75), scale='raw', nan_policy='propagate',
+           interpolation='linear', keepdims=False):
+    """
+    Compute the interquartile range of the data along the specified axis.
+
+    The interquartile range (IQR) is the difference between the 75th and 25th
+    percentile of the data. It is a measure of the dispersion similar to
+    standard deviation or variance, but is much more robust against outliers
+    [3]_.
+
+    The ``range`` parameter allows this function to compute other percentile
+    ranges than the actual IQR. For example, setting ``range=(0, 100)`` is
+    equivalent to ``numpy.ptp``.
```

Comment:
if you use single backticks for functions, they get turned into links. So that's preferred for function names.